ow_plexil/src/plexil-adapter/PlexilPlanSelection.cpp

```diff
@@ -0,0 +1,153 @@
+// The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+// Research and Simulation can be found in README.md in the root directory of
+// this repository.
+
+
+#include "PlexilPlanSelection.h"
+#include <string>
+#include <std_msgs/String.h>
+
+void PlexilPlanSelection::initialize(std::string initial_plan)
+{
+  ROS_INFO("Starting plexil node...");
+  //initialzing the Executive
+  m_executive.reset(OwExecutive::instance());
+  if (! m_executive->initialize()) {
+    ROS_ERROR("Could not initialize OW executive, shutting down.");
+    return;
+  }
+
+  //initialize the OwInterface
+  OwInterface::instance()->initialize();
+  m_genericNodeHandle = std::make_unique<ros::NodeHandle>();
+
+  // wait for the first proper clock message before running the plan
+  ros::Rate warmup_rate(0.1);
+  ros::Time begin = ros::Time::now();
+  while (ros::Time::now() - begin == ros::Duration(0.0))
+  {
+    ros::spinOnce();
+    warmup_rate.sleep();
+  }
+
+  // if launch argument plan is given we add it
+  if(initial_plan.compare("None") != 0) {
+    plan_array.push_back(initial_plan);
+  }
+  //workaround for the getPlanState not returning correctly for first plan
+  m_first_plan = true;
```

Comment:
I understand you went this route due to a problem with Plexil Executive but did you seek help on the issue? It could be a bug in PLEXIL or maybe a misunderstanding of the use of the API on our side, but either way I would suggest reaching to plexil team on `plexil-support@lists.sourceforge.net` and report the issue to them (in case you haven't already done this).

Reply:
I did reach out to Chuck about it originally, he verified that it was expected behavior and that the workaround would be necessary for now. Mike and I are talking with him to see if a new solution is necessary/what it would entail. Should I create a Jira ticket to remove the workaround in case a better Plexil solution gets added in the future?

Reply:
I think creating a Jira ticket for that is a good idea. This is a code maintainability issue that the team would need to deal with later on.